src/scales/scale.time.js

```diff
@@ -648,22 +732,31 @@ class TimeScale extends Scale {
 
 	/**
 	 * Function to format an individual tick mark
+	 * @param {number} time
+	 * @param {number} index
+	 * @param {object[]} ticks
+	 * @param {string|undefined} [format]
+	 * @return {string}
 	 * @private
 	 */
 	_tickFormatFunction(time, index, ticks, format) {
 		const me = this;
 		const options = me.options;
 		const formats = options.time.displayFormats;
+		const unit = me._unit;
 		const majorUnit = me._majorUnit;
-		const minorFormat = formats[me._unit];
-		const majorFormat = formats[majorUnit];
+		const minorFormat = unit && formats[unit];
```

Comment:
did ts want these changes? `formats[undefined]` is going to be undefined.

Reply:
Yeah, it didn't like using a possibily `undefined` variable to get the value